src/main/asciidoc/java/index.adoc

```diff
@@ -202,6 +202,69 @@ oc policy add-role-to-user view system:serviceaccount:$(oc project -q):default -
 
 Further configuration details are available on the https://github.com/jgroups-extras/jgroups-kubernetes[Kubernetes discovery protocol for JGroups repository].
 
+=== Rolling updates
+
+During rolling udpates, the Infinispan team recommends to replace pods one by one.
+
+To do so, we must configure Kubernetes to:
+
+* never start more than one new pod at once
+* forbid more than one unavailable pod during the process
+
+[source,yaml]
+----
+spec:
+  strategy:
+    type: Rolling
+    rollingParams:
+      updatePeriodSeconds: 10
+      intervalSeconds: 20
+      timeoutSeconds: 600
+      maxUnavailable: 1 <1>
+      maxSurge: 1 <2>
+----
+<1> the maximum number of pods that can be unavailable during the update process
+<2> the maximum number of pods that can be created over the desired number of pods
+
+Also, pod readiness probes must take cluster health into account.
+Indeed, when a node joins or leaves the cluster, Infinispan rebalances the data across members, and it is better to avoid concurrent state transfers.
+When the state transfer completes, the cluster health goes back to `HEALTHY`.
+
+The readiness probe can be implemented with link:../../vertx-health-check/java/[Vert.x Health Checks].
+The health check procedure should test the `DefaultCacheManager` status.
+Here's an example with the Java language:
+
+[source,java]
+----
+HealthChecks checks = HealthChecks.create(vertx)
+  .register("ispn-cluster-status", future -> {
+    VertxInternal vertxInternal = (VertxInternal) vertx; <1>
+    InfinispanClusterManager clusterManager = (InfinispanClusterManager) vertxInternal.getClusterManager(); <2>
+    DefaultCacheManager cacheManager = (DefaultCacheManager) clusterManager.getCacheContainer(); <3>
```

Comment:
Actually, you could cast it to `EmbeddedCacheManager`, see https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java#L342.

Reply:
👍